packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_model.dart

```diff
@@ -17,19 +21,27 @@ import 'logging_table_v2.dart';
 /// The [LoggingTableV2] table uses variable height rows. This model caches the
 /// relevant heights and offsets so that the row heights only need to be calculated
 /// once per parent width.
-class LoggingTableModel extends ChangeNotifier {
+class LoggingTableModel extends ChangeNotifier with DisposerMixin {
   LoggingTableModel() {
     _worker = InterruptableChunkWorker(
       callback: (index) => getFilteredLogHeight(
         index,
       ),
       progressCallback: (progress) => _cacheLoadProgress.value = progress,
     );
+
+    addAutoDisposeListener(
+      preferences.logging.retentionLimit,
+      _onRetentionLimitUpdate,
+    );
+
+    _retentionLimit = preferences.logging.retentionLimit.value;
```

Comment:
nit: set this before adding the listener to avoid the rare case where there is a race condition between the listener firing and this line getting hit